### ISSUE TITLE
Update Glossary / Methodology of work

### ISSUE THREAD

**Issue by 3imed-jaberi** [reactjs/ar.reactjs.org#122]
# Glossary 📔
---
## I think that we should update the glossary accurately, as we review all the words and the corresponding translation so as not to fall into conflict ..

# Methodology of work 🛃
---
## Should be identify contributors and provide our own communication platform such as SLACK, DISCORD, TELEGRAM or even MESSENGER ...etc , so that we can communicate in everything .. 

> Note : I know we have Slack room but the problem is that it is not active and there are some active contributors that are not in it ..

**Comment by iRayan7** [comment]
Lets do it. We will use this issue to discuss glossary changes 👍

**Comment by mohshbool** [comment]
I didn't know that there was a Slack room 😅

**Comment by 3imed-jaberi** [comment]
## Should be update the glossary in the point of ` high-order components `.
## There is a conflict between these two expressions : 
  * 1 - **المكونات ذات الترتيب الأعلى**
  * 2 - **مكونات من المرتبة الأعلى**
## ==> I prefer the first expression.

**Comment by 3imed-jaberi** [comment]
What is the proposed translation for
 * Major
 * Minor
 * Patch 
> Version : Major.Minor.Patch 

What do you think of [this site](https://semver.org/lang/ar/) as ref. !

**Comment by mohshbool** [comment]
> What is the proposed translation for
> 
> * Major
> * Minor
> * Patch
> 
> > Version : Major.Minor.Patch
> 
> What do you think of [this site](https://semver.org/lang/ar/) as ref. !

I would go with 
- Major --> رئيسي
- Minor --> ثانوي/فرعي
- Path --> رُقعة/تصحيح

**Comment by 3imed-jaberi** [comment]
There is a proposal by @Fcmam5 ( #130 )  to amend the translation of **[DOCS](https://ar.reactjs.org/docs/getting-started.html)** from `المستندات` to `التوثيق` ..

**Comment by mohshbool** [comment]
> There is a proposal by @Fcmam5 ( #130 ) to amend the translation of **[DOCS](https://ar.reactjs.org/docs/getting-started.html)** from `المستندات` to `التوثيق` ..

I would keep it as is, its clearer that way imo

**Comment by 3imed-jaberi** [comment]
What is the proposed translation for ; `Data-Driven` .. ?

**Comment by 3imed-jaberi** [comment]
What is the proposed translation for ; `Data-Driven` .. ?

**Comment by Aissaoui-Ahmed** [comment]
<div dir="rtl">
مرحبا بالجميع <BR>
لدي إقتراح أتنمى أن ينال القبول من الجميع بخصوص القاموس وترجمة الكلمات التقنية الخاصة بالبرمجة، لكل منا جهود خاصة في الترجمة ومحاولة إثراء المحتوى العربي عموما، إلا أنه قد يجد إختلاف مع باقي المبرمجين ولا أعتقد أن  كل واحد منا قد مر بهذا الموقف وفي محاولة لتفادي الصدام أنشأت مستودع مع زميلي @celyes يخص الكلمات المتعلقة بالبرمجة ليكون مرجع نستند إليه.<br>
هذا المشروع قيد الإنجاز لذا تترددو في طرح إقتراحات قد تساعد في تطويره.<br>
معا تحياتي للجميع ❤
</div>

repositorie [i18n](https://github.com/softvenue/i18n)

**Comment by celyes** [comment]
لتفادي الاختلاط في المصطلحات التقنية، يرجى منكم ان الاطلاع على هذا المستودع:
[SoftVenue i18n](https://github.com/softvenue/i18n)
كما أن أي مساهمة فيه مرحب بها و تحياتنا للجميع
@SoftVenue

**Comment by Aissaoui-Ahmed** [comment]
just suggestions

bug = خطأ
framework = أداة 
log = سجل
release = إصدار
bind = ربط

**Comment by Fcmam5** [comment]
> just suggestions
> 
> bug = خطأ
> framework = أداة
> log = سجل
> release = إصدار
> bind = ربط

Hi Ahmed, some of these terms have other translations that everyone is using like:
bug = عِلّة or خطأ برمجي
framework = إطار عمل

And these ones seems to be the right translations:
> log = سجل
> release = إصدار
> bind = ربط

**Comment by 3imed-jaberi** [comment]
The last three words shouldn't be translated .. Leave it as is ...

**Comment by Fcmam5** [comment]
Style Guide = دليل التنسيق

**Comment by 3imed-jaberi** [comment]
> I suggest `دليل التنسيق` as a translation for `Style Guide`

@Fcmam5 
please bro , add this to glossary !

**Comment by Fcmam5** [comment]
TLDR;
Facebook => فيسبوك or Facebook
Instagram => Instagram
Web => ويب

Please translate `Facebook` as `فيسبوك` or keep it as it is. I know sometimes it causes some formatting issues if we keep it untranslated.

I found that it's translated as `فيس بوك` [here](https://ar.reactjs.org/community/conferences.html#react-native-eu-2019) which is not correct (please check Facebook's arabic pages)

َAlso, page I found that `Instagram` is translated as `انستقرام` which is incorrect, in [Facebook's help pages](https://www.facebook.com/help/instagram/111923612310997?helpref=related) the word `Instagram` is not translated, so there's no official translation for the word, for that we should keep it as it is.

**Comment by 3imed-jaberi** [comment]
Instagram ==> انستقرام or Instagram or [check this for more details](https://ar.wikipedia.org/wiki/%D8%A5%D9%86%D8%B3%D8%AA%D8%BA%D8%B1%D8%A7%D9%85)
Facebook ==> فيس بوك or فيسبوك or Facebook or [check this for more details](https://ar.wikipedia.org/wiki/%D9%81%D9%8A%D8%B3%D8%A8%D9%88%D9%83)

Prove of FB: 
![image](https://user-images.githubusercontent.com/43971542/62251891-7a308080-b3f1-11e9-928c-e339b3b47c20.png)

**Comment by 3imed-jaberi** [comment]
> TLDR;
> Facebook => فيسبوك or Facebook
> Instagram => Instagram
> Web => ويب
> 
> Please translate `Facebook` as `فيسبوك` or keep it as it is. I know sometimes it causes some formatting issues if we keep it untranslated.
> 
> I found that it's translated as `فيس بوك` [here](https://ar.reactjs.org/community/conferences.html#react-native-eu-2019) which is not correct (please check Facebook's arabic pages)
> 
> َAlso, page I found that `Instagram` is translated as `انستقرام` which is incorrect, in [Facebook's help pages](https://www.facebook.com/help/instagram/111923612310997?helpref=related) the word `Instagram` is not translated, so there's no official translation for the word, for that we should keep it as it is.

Instagram ==> انستقرام or Instagram or [check this for more details](https://ar.wikipedia.org/wiki/%D8%A5%D9%86%D8%B3%D8%AA%D8%BA%D8%B1%D8%A7%D9%85)
Facebook ==> فيس بوك or فيسبوك or Facebook or [check this for more details](https://ar.wikipedia.org/wiki/%D9%81%D9%8A%D8%B3%D8%A8%D9%88%D9%83)

Prove of FB: 
![image](https://user-images.githubusercontent.com/43971542/62251891-7a308080-b3f1-11e9-928c-e339b3b47c20.png)

**Comment by 3imed-jaberi** [comment]
dependencies ==> إعتمادات

**Comment by mohdule** [comment]
How should i translate `continuous integration`, should i translate it ?

**Comment by asantarissy** [comment]
> How should i translate `continuous integration`, should i translate it ?

I did a quick research and found the following
* تركيب متواصل [Wikipedia][wiki]
* التكامل المستمر [meemapps.com][meemapps.com], [cloudworld.io][cloudworld.io]

Which one you think we should use?

[wiki]: https://ar.wikipedia.org/wiki/%D8%AA%D8%B1%D9%83%D9%8A%D8%A8_%D9%85%D8%AA%D9%88%D8%A7%D8%B5%D9%84
[meemapps.com]: https://www.meemapps.com/term/5a3da4e26178b74f704a4a80/Continuous%20Integration-التكامل%20المتواصل
[cloudworld.io]: https://cloudworld.io/التكامل-المستمر-continuous-integration/

**Comment by 3imed-jaberi** [comment]
I think `التكامل المستمر` is better .. !

**Comment by mohdule** [comment]
So, `mocking`
I translated it to `محاكاة`

لا ادري اذا كان هنالك كلمة أفضل
if it's ok, i think it should be added to the glossary
to help other people working on the testing pages

**Comment by 3imed-jaberi** [comment]
Timer ?! 
مؤقتات or عداد 🤔 !!

**Comment by Fcmam5** [comment]
Please, what translation should we use for "Concurrent mode" ?

- وضع متزامن
- وضع تنافسي

For the first suggestion it was applied by Arabyes team: https://www.arabeyes.org/techdict:Concurrent, and in some Hsoub translations as `المهام الفرعية (Threads، وتدعى "الخيوط" أيضًا) هي الطريقة التي تنفذ بها روبي البرمجة المتزامنة (concurrent programming). البرامج التي تتطلب تنفيذ عدة مهام فرعية يمكنها استخدام الصنف Thread.` in https://wiki.hsoub.com/Ruby
And the second suggestion seems also logical and it was chosen in this blog:
https://www.tutomena.com/javascript/react-concurrent-mode/

What's your input?

**Comment by Aissaoui-Ahmed** [comment]
<div dir=rtl>
أقتراح 'وضع توافقي'

**Comment by atefBB** [comment]
<div dir="rtl">أقترح استعمال مصطلح "خادوم" للتعبير عن الـserver ﻷن هذه الكلمة على وزن فاعول الذي يأتي في اللغة العربية ﻷسماء اﻵلات ومنه "قادوم" واستعمل حديثا في ترجمة "حاسوب".</div>

**Comment by atefBB** [comment]
<div dir="rtl">اختصارا للمجهود وحسما لاختلاف الترجمات، اقترح استعمال القاموس التقني لمشروع عرآبياز (<a href="https://www.arabeyes.org/%D8%A7%D9%84%D9%82%D8%A7%D9%85%D9%88%D8%B3_%D8%A7%D9%84%D8%AA%D9%82%D9%86%D9%8A">الرابط</a>)، فقد بُذل فيه مجهود وهو شامل تقريبا.</div>
@3imed-jaberi @Aissaoui-Ahmed @celyes واﻵخرين

**Comment by MaherAlshaar** [comment]
> لتفادي الاختلاط في المصطلحات التقنية، يرجى منكم ان الاطلاع على هذا المستودع:
> [SoftVenue i18n](https://github.com/softvenue/i18n)
> كما أن أي مساهمة فيه مرحب بها و تحياتنا للجميع
> @softvenue

i hope you all the Best

**Comment by 3imed-jaberi** [comment]
Fallback ===> fallback or احتياطي or  both `احتياطي (fallback)` ??

ping @reactjs/reactjs-org-arabic-translation !!

**Comment by A7med3bdulBaset** [comment]
What is the proposed translation for `code`?

**Comment by 3imed-jaberi** [comment]
> What is the proposed translation for `code`?

كود

**Comment by shadigaafar** [comment]
> I suggest `دليل التنسيق` as a translation for `Style Guide`

كلمة style يجب ترجمتها بحذر، ومحاولة للتوفيق بين الترجمات المتوافرة الموجودة مسبقا، الكثير من الشركات نفس ميكروسوفت وجوجل، يترجموها "نمط".

**Comment by shadigaafar** [comment]
وأهم شيء، تبقى ترجمة render هي "تصيير".

**Comment by shadigaafar** [comment]
> bug = عِلّة or خطأ برمجي

من الممكن استخدام الترجمتين معًا، لأنه في حالة استخدام "علّة" لأن يفهم البعض، لكن يمكن توضيحها بين قوسين كـ (خطأ برمجي)، حتى يعتاد عليها الناس.

**Comment by shadigaafar** [comment]
> * Major --> رئيسي
> * Minor --> ثانوي/فرعي
> * Path --> رُقعة/تصحيح

لماذا لا تكون "ترقيع"

**Comment by shadigaafar** [comment]
> dependencies ==> إعتمادات

اعتمادات من غير همزة

**Comment by shadigaafar** [comment]
بالنسبة لترجمة community، فهذا صعبة، ولا أحبذ ترجمتها "مجتمع"، أرى أن التمييز بين society وcommunity مهم للغاية.  الكثير يعتقد أنه ليس هناك كلمة عربية مقابلة لـ community، فلذلك يترجموها society، في حين أن هناك مصطلح مستحدث أتي به كمال أبو ديب، وهو استاذ جامعي كبير يدرس في جامعة لندن، وهذا المصطلح المستحدث هو "**منجمع**"، يستخدمه كثيرا في كتبه، وقد تم تسجيله أيضا في ويكاموس، يعني يستطيع الناس الرجوع له لمعرفة معناه. الرجاء الاطلاع على الرابط[هنا](https://ar.wiktionary.org/wiki/%D9%85%D9%8F%D9%86%D9%92%D8%AC%D9%8E%D9%85%D9%8E%D8%B9). لا يمكن للعربية أن تطور من غير القليل من الجراءة في استخدام المصطلحات المستحدثة. في الإنجليزية تظهر مصطلحات جديدة كل يوم. بينما العرب يرفضون أي جديد.  أعرف أن البعض لديه رُهاب الكلمات المستحدثة، لكن بمجرد أن توضع الكلمة موضع الاستخدام، ستصبح كلمة عادية. وقد يكون هذا على يدكم. وشكرا لجهودكم!

**Comment by shadigaafar** [comment]
اشكالية أخرى قد تواجها هو ترجمة مصطلح array، الكثير من الترجمات تقول "مصفوفة"، بينما لغويًا معبرة، لكن قد تتعارض مع مصطلح matrix المستخدام أيضا في البرمجة بمعنى "مصفوفة" أيضا، كما أن كتب الرياضيات تترجمة نفسه الترجمة. فلذلك أنه لا بد من صناعة تمييز. واقترح ترجمتها إلى "نظم".

**Comment by atefBB** [comment]
> > What is the proposed translation for `code`?
> 
> كود

اقترح لفظ "شفرة" أو "شيفرة" لشهرتهما فقط، لغويا يمكن أن يعتبر اللفظ تعريبا وهو نقل لفظ أعجمي إلى العربية مع استعمال الصوتيات المتوفرة في اللغة العربية ومع احترام أنظمة اللغة العربية.

**Comment by atefBB** [comment]
> > bug = عِلّة or خطأ برمجي
> 
> من الممكن استخدام الترجمتين معًا، لأنه في حالة استخدام "علّة" لن يفهم البعض، لكن يمكن توضيحها بين قوسين كـ (خطأ برمجي)، حتى يعتاد عليها الناس.

أعتقد أن خطأ برمجي أوضح تعبيريا. أظن أن الازدواجية في الترجمة تخلق لبسا مستداما خاصة أن هناك من سيتربى على وجود المصطلحين

**Comment by atefBB** [comment]
> بالنسبة لترجمة community، فهذا صعبة، ولا أحبذ ترجمتها "مجتمع"، أرى أن التمييز بين society وcommunity مهم للغاية. الكثير يعتقد أنه ليس هناك كلمة عربية مقابلة لـ community، فلذلك يترجموها society، في حين أن هناك مصطلح مستحدث أتي به كمال أبو ديب، وهو استاذ جامعي كبير يدرس في جامعة لندن، وهذا المصطلح المستحدث هو "**منجمع**"، يستخدمه كثيرا في كتبه، وقد تم تسجيله أيضا في ويكاموس، يعني يستطيع الناس الرجوع له لمعرفة معناه. الرجاء الاطلاع على الرابط [هنا](https://ar.wiktionary.org/wiki/%D9%85%D9%8F%D9%86%D9%92%D8%AC%D9%8E%D9%85%D9%8E%D8%B9). لا يمكن للعربية أن تتطور من غير القليل من الجراءة في استخدام المصطلحات المستحدثة. في الإنجليزية تظهر مصطلحات جديدة كل يوم. بينما العرب يرفضون أي جديد. أعرف أن البعض لديه رُهاب الكلمات المستحدثة، لكن بمجرد أن توضع الكلمة موضع الاستخدام، ستصبح كلمة عادية. وقد يكون هذا على يدكم. وشكرا لجهودكم!

اعذرني لكن أظن أن المصطلح ركيك، ما الغرابة في ترجمة مجتمع وفيه ما يحقق المعنى في اللغة اﻷنجليزية.

**Comment by atefBB** [comment]
> الاشكالية الأخرى التي قد تواجهوها هي ترجمة مصطلح array، الكثير من الترجمات تقول "مصفوفة"، في حين أنها لغويًا معبرة، لكن قد تتعارض مع مصطلح matrix المستخدام أيضا في البرمجة بمعنى "مصفوفة" أيضا، كما أن كتب الرياضيات تترجمه نفس الترجمة. فلذلك لا بد من صناعة تمييز. واقترح ترجمتها إلى "نظم".

أليست ترجمة نظم لشيء أعم مثل systems ؟ هنا يحدث التباس -حسب ظني- بين المعنيين (أقصد martrix  و systems (

**Comment by shadigaafar** [comment]
> أليست ترجمة نظم لشيء أعم مثل systems ؟ هنا يحدث التباس -حسب ظني- بين المعنيين (أقصد martrix و systems (

يبدو أن هناك سوء فهم، اقصد "نَظْم"، وليس "نُظم". كما أن ترجمة system نظام وجمها أنظمة.

**Comment by shadigaafar** [comment]
> اعذرني لكن أظن أن المصطلح ركيك، ما الغرابة في ترجمة مجتمع وفيه ما يحقق المعنى في اللغة اﻷنجليزية.

أنا لا أرى فيه أي ركاكة على الاطلاق. يبدو فقط لأنك غير معتاد عليها. كما أنه أتى، كما ذكرت في الأعلى، من شخص حجة في اللغة. ثانيا: هناك فرق بين society وcommunity، وكل اللغات تميز بينها، بل وحتى الفارسية. التمييز موجود في كل اللغات. المجتمع أوسع، والمنجمع أضيق. المجتمع هو مجموعة من الناس **يشتركون في خصائص ثقافية واجتماعية واقتصادية.** والمنجمع هو مجموعة من الأفراد الذين يعيشون في موقع جغرافي محدد و**يشتركون في مصلحة أو هدف أو غرض مشترك**. المجتمع أوسع وأكثر تجريدًا، على عكس المنجمع. ثالثا: إذا ظلينا على هذه العقلية،  أقصد عقلية رفض التحديث في اللغة والمصطلحات المستحدثة، فنحن نخلق لغة ككل ركيكة وغير صالحة أكاديميًا، في النهاية سينفر الناس منها لأنها لغة غير دقيقة. فالخيار لكم، مصطلح "ركيك" كما تقول، أو لغة غير دقيقة. أنا قلت رأي وما عندي. ولست هنا لأجبر أحد على شيء.

**Comment by atefBB** [comment]
> > أليست ترجمة نظم لشيء أعم مثل systems ؟ هنا يحدث التباس -حسب ظني- بين المعنيين (أقصد martrix و systems (
> 
> يبدو أن هناك سوء فهم، اقصد "نَظْم"، وليس "نُظم". كما أن ترجمة system نظام وجمها أنظمة.

نعم نظم بفتح النون فصيحة ومعبرة على المعنى.

**Comment by atefBB** [comment]
> > اعذرني لكن أظن أن المصطلح ركيك، ما الغرابة في ترجمة مجتمع وفيه ما يحقق المعنى في اللغة اﻷنجليزية.
> 
> أنا لا أرى فيه أي ركاكة على الاطلاق. يبدو فقط لأنك غير معتاد عليها. كما أن المصطلح أتى، كما ذكرت في الأعلى، من شخص حجة في اللغة. ثانيا: هناك فرق بين society وcommunity، وكل اللغات تميز بينها، بل وحتى الفارسية. التمييز موجود في كل اللغات. المجتمع أوسع، والمُنجمع أضيق. المجتمع هو مجموعة من الناس **يشتركون في خصائص ثقافية واجتماعية واقتصادية.** والمنجمع هو مجموعة من الأفراد الذين يعيشون في موقع جغرافي محدد و**يشتركون في مصلحة أو هدف أو غرض مشترك أو خاصية مشتركة**. المجتمع أوسع وأكثر تجريدًا، على عكس المنجمع. فعلى سبيل المثال، حينما نقول "منجمع رياكت" نقصد جماعة من الناس تشترك في غرض وهدف وخاصية مشتركة، مثل تطوير رياكت. ثالثا: إذا ظلينا على هذه العقلية، أقصد عقلية رفض التحديث في اللغة والمصطلحات المستحدثة، فنحن نخلق لغة ككل ركيكة وغير صالحة أكاديميًا، في النهاية سينفر الناس منها لأنها لغة غير دقيقة، إذا لم يكن هذا قد حدث بالفعل. فالخيار لكم، مصطلح "ركيك" كما تقول، أو لغة غير دقيقة. أنا قلت رأي وما عندي. ولست هنا لأجبر أحد على شيء.

أظن أن الموضوع سيطول النقاش فيه، أنا شخص ضد ما يسمى التجديد اللغوي بلا ضابط، وهو حقيقة موضوع شائك وأسال حبرا ومازال يسيل، الصد المنيع للغة العربية هي لغة القرآن التي نستمد منها التجديد والتحديث غير ذلك سيحدث فوضى كالتي نحن فيها، يأتي شخص مغمور فينبثق لسانه على لفظ استحسنه فيغري به من يشاء فيصبح مرجعا. أعذرني على الكلام القاسي لكني لا أقصدك شخصيا. منجمع لا يستقيم على أوزان اللغة فوزن منفعل له معانيه التي ضبطتها كتب اللغة والمعاجم. اﻷوزان ومعانيها تسير معا وهي مما تواضع عليه العرب من قديم ولا سبيل للتجديد فيه -حسب رأيي-، المرجع هم أهل اللغة أولا وأخيرا ولا أدري هل الاستاذ الجامعي المشار إليه منهم أو لا.

**Comment by shadigaafar** [comment]
> أظن أن الموضوع سيطول النقاش فيه، أنا شخص ضد ما يسمى التجديد اللغوي بلا ضابط، وهو حقيقة موضوع شائك وأسال حبرا ومازال يسيل، الصد المنيع للغة العربية هي لغة القرآن التي نستمد منها التجديد والتحديث غير ذلك سيحدث فوضى كالتي نحن فيها، يأتي شخص مغمور فينبثق لسانه على لفظ استحسنه فيغري به من يشاء فيصبح مرجعا. أعذرني على الكلام القاسي لكني لا أقصدك شخصيا. منجمع لا يستقيم على أوزان اللغة فوزن منفعل له معانيه التي ضبطتها كتب اللغة والمعاجم. اﻷوزان ومعانيها تسير معا وهي مما تواضع عليه العرب من قديم ولا سبيل للتجديد فيه -حسب رأيي-، المرجع هم أهل اللغة أولا وأخيرا ولا أدري هل الاستاذ الجامعي المشار إليه منهم أو لا.

ليس من الضروي أن يكون النقاش طويلاً؛ فهو مجرد اقتراح بسيط. لا توجد مشكلة في استخدام الوزن "مُنْجَمَع"، لأنه اسم مفعول من الفعل "انجَمَع". بالنسبة للأستاذ الذي تم ذكره سابقاً، فهو ليس مجرد أستاذ جامعي بل هو **رئيس قسم اللغة العربية في جامعة لندن**. فيما يتعلق بالقرآن، لا أرى أي صلة بينه وبين الموضوع المطروح، لذا لا أرى ضرورة لذكره هنا. من الأفضل أن يتم التجديد داخل اللغة نفسها بدلاً من الاعتماد على الألفاظ الأجنبية. لا يوجد حديث عن تجديد حقيقي هنا؛ بل هو إضافة وإثراء للغة، وليس تجديدًا يتضمن تغيير قواعد وهيكلة اللغة. مع احترامي الشديد لك، لا أرى سببًا واضحًا لحالة الهلع التي تمالكتك. الاستمرار في استخدام نفس الألفاظ والكلمات وإضافة معاني جديده لها كل مرة دون محاولة إثراء اللغة لن يكون له إلا عواقب وخيمة على اللغة، كأن يهجرها أهلها، هذا إذا لم يكن قد حدث بالفعل.

**Comment by shadigaafar** [comment]
> What is the proposed translation for ; `Data-Driven` .. ?

**مُستنِد إلى البيانات**، هذه هي ترجمة الرسمية المستخدمة في مواقع جوجل بالعربي.

**Comment by shadigaafar** [comment]
> اختصارا للمجهود وحسما لاختلاف الترجمات، اقترح استعمال القاموس التقني لمشروع عرآبياز ([الرابط](https://www.arabeyes.org/%D8%A7%D9%84%D9%82%D8%A7%D9%85%D9%88%D8%B3_%D8%A7%D9%84%D8%AA%D9%82%D9%86%D9%8A))، فقد بُذل فيه مجهود وهو شامل تقريبا.
> @3imed-jaberi @Aissaoui-Ahmed @celyes واﻵخرين

لا أعتقد أن "عرآبياز" يعد مرجعًا موثوقًا للمصطلحات التقنية أو مصدرًا لغويًا معتمدًا. فعلى الرغم من أنه قد يكون له دور في تقديم بعض الترجمات، إلا أن هناك العديد من الأخطاء والهفوات التي يمكن ملاحظتها في ترجماته. على سبيل المثال، يستخدم "عرآبياز" في الغالب فعل الأمر في سياقات حاسوبية بطريقة غير ملائمة، وذلك بدلًا عن المصدر، كما هو شائع. 

كذلك، نجد أن الترجمات التي يقدمها لبعض المصطلحات التقنية ليست دائمًا دقيقة. فعلى سبيل المثال، يقومون بترجمة المصطلح "select" إلى "اختر" أو "اختيار"، بينما الترجمة الأكثر دقة والأكثر شيوعًا في المجال الحاسوبي هي "تحديد". من المهم التمييز بين "تحديد" و"اختيار"؛ حيث أن "اختيار" تُستخدم عادة كترجمة لـ "choose". هناك حالات استثنائية لترجمة select إلى اختيار، لكن ليس دائما. 

بناءً على ذلك، من الأفضل الاعتماد على مصادر أخرى ومتخصصة في مجال التقنية لضمان دقة وموثوقية الترجمات المتعلقة بالمصطلحات التقنية.

**Comment by shadigaafar** [comment]
بالنسبة إلى ترجمة مصطلح "`stack`"، اقترح ترجمتها إلى بـ "كُدس" ، حيث تعكس مفهوم التراكم العمودي للأشياء. ومصدرها "تكديس"، يقال في العربية، كدس الأشياء بعضها فوق بعض، أي راكمها بشكل عمودي لا أفوقي. وتعبر بشكل جيد عن طبيعة الهيكل البياني الذي يشير إلى ترتيب العناصر بشكل تراكمي عمودي. هذا الاستخدام معتمد أيضًا في ترجمات ميكروسوفت، كما أنه لا بديل غيره.

**Comment by shadigaafar** [comment]
اقترح ترجمة مصطلح "stack" إلى **كُدس** كترجمة دقيقة، حيث تعكس مفهوم التراكم العمودي للأشياء. ومصدرها "تكديس"، يقال في العربية: كدس الأشياء بعضها فوق بعضها، أي وضعها فوق بعضها البعض (عموديًا). وتعبر بشكل جيد عن طبيعة الهيكل البياني الذي يشير إلى ترتيب العناصر بشكل تراكمي. كما يعد هذا المصطلح معتمدًا في ترجمات ميكروسوفت.
